test/isFocusable.jest.test.js

```diff
@@ -0,0 +1,283 @@
+const { getByTestId, getByText } = require('@testing-library/dom');
+const { isFocusable } = require('../src/index.js');
+
+describe('isFocusable', () => {
+  it('throws an error if no node is provided', () => {
+    expect(() => isFocusable()).toThrow();
+  });
+
+  describe('returns true', () => {
+    it('returns true for a `button` element', () => {
+      const container = document.createElement('div');
+      container.innerHTML = '<button>Click me</button>';
+
+      expect(isFocusable(getByText(container, 'Click me'))).toBe(true);
+    });
+
+    it('returns true for an `input` element', () => {
+      const container = document.createElement('div');
+      container.innerHTML = '<input data-testid="testInput" />';
+
+      expect(isFocusable(getByTestId(container, 'testInput'))).toBe(true);
+    });
+
+    it('returns true for a `select` element', () => {
+      const container = document.createElement('div');
+      container.innerHTML = `<select data-testid="testSelect">
+          <option value="foo">foo</option>
+        </select>`;
+
+      expect(isFocusable(getByTestId(container, 'testSelect'))).toBe(true);
+    });
+
+    it('returns true for a `textarea` element', () => {
+      const container = document.createElement('div');
+      container.innerHTML = '<textarea data-testId="testTextarea"></textarea>';
+
+      expect(isFocusable(getByTestId(container, 'testTextarea'))).toBe(true);
+    });
+
+    it('returns true for an `a` anchor element with an `href` attribute', () => {
+      const container = document.createElement('div');
+      container.innerHTML =
+        '<a href="https://github.com/focus-trap/tabbable">Focusable</a>';
+
+      expect(isFocusable(getByText(container, 'Focusable'))).toBe(true);
+    });
+
+    // The README says `isFocusable` should return `true for:
+    // <a> elements with href or xlink:href attributes
+    // But this is returning `false`.
+    // Should the README be updated to not include `xlink:href` here?
+    it.skip('returns true for an `a` anchor element with an `xlink:href` attribute', () => {
```

Comment:
I need to confirm this behavior manually in a real app, outside of a test suite.

Reply:
Like @idoros suggests, I think adding `a[xlink:href]` to the `candidateSelectors` list in the code would solve this. We should fix the code and leave the docs as they are (although it would've been like this for a _long_ time now, so if anyone was dependent on `<a xlink:href>` being tabbable, we should've heard by now, unless they created their own fork and fixed it there while the repo was unmaintained).

Reply:
Thanks @stefcameron and @idoros, mostly I was wondering how you guys would like to tackle it. To me, it seems like the package currently _does not_ consider `a[xlink:href]` to be tabbable, like you said.

I'd probably recommend removing the line from the README first as part of this PR, and then maybe in a followup PR add the new line of code in the `candidateSelectors` array and put that line back in the README. That way the PR stays nice and focused on the test suite as opposed to also adding in new functionality to support `a[xlink:href]`.

You good with that approach? I'm fine either way since it's a small change, it just seems like a better approach to me to keep the PR focused to a single thing.

Reply:
Well, looking at the Blame for `index.js` where the selectors are defined, it hasn't changed in over 3 years except 4 months ago when we added support for `<details>` elements. So IF `<a xlink:href>` had ever been supported at some point, it stopped being supported over 3 years ago. Going back to 5-6 years ago in the Blame, I still don't see `xlink` anywhere in `index.js`.

So I'm thinking it's safe to remove it from the README, but now we should include a Changeset with this change just to point out that we've removed adjusted the README to technically remove a feature, but that the code never actually supported it. Just for transparency. Technically, it would be breaking and so a major, but since it looks like it was never supported, perhaps a minor bump would be OK. I'd like to avoid a major bump in any case.

In light of all this history, I guess fixing the bug could be just as potentially problematic -- suddenly all these apps that for some reason use `<a xlink:href>` and `tabbable` now suddenly have all these new tabbable nodes and it messes everything up. Could be bad that way too.

I'm leaning on the side of removing from the README and adding a Changeset for a minor bump with a "heads-up" note.

Reply:
Awesome, that sounds like a good plan to me. I'll remove that line from the README and add a changeset for this MR as a minor bump.

Reply:
Actually now that I'm working on this, just to clarify, this would be a `patch` version as opposed to a `minor` version. Because this would be more like a "fix" than a new "feature" being added.

Reply:
Yeah, I was debating between `patch` and `minor`. My thinking was that since `minor` is for new functionality, and we're _removing_ functionality (that hasn't been in place for at least 6 years, AFAICT), it's like the same "level" or "type" of change. If we consider it a bug in the README and we just want to inform people, then I a `patch` would work. I thought a `minor` was a stronger signal that some significant change had taken place (also warrants a `minor` per semver.org; not just new features, but significant change in behavior -- that's still backward-compatible), without going to the level of `major`.

I guess it's a fine line. :)